### ISSUE TITLE
public went walkabouts on colobus

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/MyopicVicar#1049]
At 00:30 GMT on Jan 3, 2016 i did a restart.txt on coloubus to change a sleep parameter in mongo_config.yml.
On accessing colobus I found that the css was missing.
An Apache restart showed that the public directory was missing.
Stopped server access for an hr while conducting  investigation.
Rebuilt the public assets. Did not include anything else.
Server back in production.
Further clean up will be required later in the say after some sleep

**Comment by Captainkirkdawson** [comment]
Neither Doug nor I can see any reason for such an event that appears to require a deliberate removal of the folder. 
None of our scripts involve such an action. It is not affected by git actions so likely required either human intervention or hardware malfunction.
Screenshot uploaded to contact and feedback have been lost.
Will take colobus out of production for FR while completing recovery.

**Comment by Captainkirkdawson** [comment]
Log investigation shows that problem occurred after a passenger spawning error
"Cannot checkout session because a spawning error occurred. The identifier of the error is 81c3b341. Please see earlier logs for details about the error."

**Comment by dougkdev** [comment]
Apache logs narrow down the timeframe that public went missing to between 20:48:16 GMT and 20:52:15 GMT on 02 Jan 2017 (probably in the 30-second window between  20:50:47 and 20:51:21).

**Comment by dougkdev** [comment]
I don't see anything in the auth.log that would point to any sudo commands that would have caused a problem in that time frame.

**Comment by dougkdev** [comment]
What time was the passenger error?

**Comment by Captainkirkdawson** [comment]
Was a Red Herring

According to the passenger log it was OK at 2017-01-02 20:51:19 +0000 and
bad at just before 2017-01-02 20:51:21 +0000 this is a solid 3 hr before I
saw the issue and it clearly was nothing to do with the restart. It was
already gone.

What happened in between was an update of a refinery page

App 31799 stdout: Completed 200 OK in 22ms (Views: 10.3ms | ActiveRecord:
1.2ms)
App 31799 stdout: Started GET
"/search_queries/new?search_id=586abd40f493fd7c37968d41" for 5.9.125.163 at
2017-01-02 20:51:19 +0000
App 31799 stdout: Processing by SearchQueriesController#new as HTML
App 31799 stdout:   Parameters: {"search_id"=>"586abd40f493fd7c37968d41"}
App 31799 stdout:   Rendered search_queries/_form.html.erb (106.8ms)
App 31799 stdout:   Rendered search_queries/new.html.erb within
layouts/application (107.1ms)
App 31799 stdout:   Rendered layouts/_main_nav.html.erb (3.1ms)
App 31799 stdout:   Rendered layouts/_header.html.erb (0.1ms)
App 31799 stdout:   Rendered layouts/_footer.html.erb (0.6ms)
App 31799 stdout: Completed 200 OK in 125ms (Views: 112.6ms | ActiveRecord:
1.3ms)


App 31799 stdout: Started POST "/search_queries" for 5.9.125.163 at
2017-01-02 20:51:20 +0000
App 31799 stdout: Processing by SearchQueriesController#create as HTML
App 31799 stdout:   Parameters: {"utf8"=>"✓",
"authenticity_token"=>"WxA46x4jVwX2ra+kFt2DtVV53WiONfr0VxfUsn6oCzTaNa8ZjdYQWTpYIBQIMHHwqPI2uXUMDkSqQwdt+S8ewg==",
"search_query"=>{"last_name"=>"Martin", "first_name"=>"Ernest",
"start_year"=>"1917", "end_year"=>"1917", "chapman_codes"=>["", "STS"],
"place_ids"=>["5583614ef493fdf0f200169b"], "record_type"=>"bu",
"region"=>""}, "button"=>""}
App 31799 stdout: FREEREG:SEARCH_HINT: place_ln_fn
App 714 stdout: Started GET
"/cms/information-for-transcribers/creating-a-transcription-file" for
5.9.125.163 at 2017-01-02 20:51:20 +0000
App 714 stdout: Processing by Refinery::PagesController#show as HTML
App 714 stdout:   Parameters:
{"path"=>"information-for-transcribers/creating-a-transcription-file",
"locale"=>:en}
App 31162 stderr:
App 31162 stderr: -- SANITIZED CONTENT WARNING --
App 31162 stderr: Refinery::Pages::SectionPresenter#wrap_content_in_tag
App 31162 stderr: HTML attributes and/or elements content has been sanitized
App 31162 stderr:
App 31162 stderr:  <style scoped="scoped">
App 31162 stderr:    abbr { text-decoration: none; }
App 31162 stderr:    figcaption { font-style: italic; font-size: smaller; }
App 31162 stderr:  </style>
App 31162 stderr:
App 31162 stderr:  <div class="grid">
App 31162 stderr:    <aside class="grid__item one-third palm-one-whole
float--right">
App 31162 stderr:
App 31162 stderr:      <div class="islet islet--bordered">
App 31162 stderr:        <h2 class="title-block weight--semibold"
style="color: #657275;">On this page</h2>
App 31162 stderr:        <ul class="sub-nav">
App 31162 stderr: [31m-        <li><a href="#winReg">WinFreeREG</a> </li>
[0m
App 31162 stderr: [31m-        <li><a href="#spreadsheet">Spreadsheet</a>
</li> [0m
App 31162 stderr: [31m-        <li><a href="#database">Database</a> </li>
[0m
App 31162 stderr: [31m-        <li><a href="#textEditor">Text editor</a>
</li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a href="#winReg">WinFreeREG</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a href="#spreadsheet">Spreadsheet</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a href="#database">Database</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a href="#textEditor">Text editor</a> </li> [0m
App 31162 stderr:        </ul>
App 31162 stderr:      </div>
App 31162 stderr:
App 31162 stderr:      <div class="islet islet--bordered">
App 31162 stderr: [31m-      <h3 class="title-block"><a
href="/cms/information-for-transcribers">Information for transcribers</a>
</h3> [0m
App 31162 stderr: [32m+      <h3 class="title-block"> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers">Information for transcribers</a>
</h3> [0m
App 31162 stderr:        <ul class="sub-nav">
App 31162 stderr: [31m-        <li class="active"><a
href="/cms/information-for-transcribers/creating-a-transcription-file"
title="WinREG, WINFreeReg, spreadsheet, database, text editor">Creating a
transcription file</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/entering-data-from-registers"
title="What to put where for baptisms, marriages and burials">Entering data
from registers</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/dealing-with-uncertainty"
title="Viewing and reading registers; recording the un-readable">Dealing
with uncertainty</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/numbers-and-dates" title="English
and Latin, numerals and words; old-style and new-style dates, Regnal
years">Numbers and dates</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/chapman-codes" title="All the
3-letter county codes">Chapman codes</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/file-and-batch-actions"
title="Uploading/Replacing files, editing/managing batches online">File and
Batch actions</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/file-headers" title="How to
describe your file contents - spreadsheet, database, text editor">File
headers</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/permission-to-transcribe"
title="Ensure your source can be used for FreeReg">Permission to
transcribe?</a> </li> [0m
App 31162 stderr: [31m-        <li><a
href="/cms/information-for-transcribers/latin-words-to-recognise"
title="Latin words used in older registers">Latin words to recognise</a>
</li> [0m
App 31162 stderr: [32m+        <li class="active"> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/creating-a-transcription-file"
title="WinREG, WINFreeReg, spreadsheet, database, text editor">Creating a
transcription file</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/entering-data-from-registers"
title="What to put where for baptisms, marriages and burials">Entering data
from registers</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/dealing-with-uncertainty"
title="Viewing and reading registers; recording the un-readable">Dealing
with uncertainty</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/numbers-and-dates" title="English
and Latin, numerals and words; old-style and new-style dates, Regnal
years">Numbers and dates</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/chapman-codes" title="All the
3-letter county codes">Chapman codes</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/file-and-batch-actions"
title="Uploading/Replacing files, editing/managing batches online">File and
Batch actions</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/file-headers" title="How to
describe your file contents - spreadsheet, database, text editor">File
headers</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/permission-to-transcribe"
title="Ensure your source can be used for FreeReg">Permission to
transcribe?</a> </li> [0m
App 31162 stderr: [32m+        <li> [0m
App 31162 stderr: [32m+<a
href="/cms/information-for-transcribers/latin-words-to-recognise"
title="Latin words used in older registers">Latin words to recognise</a>
</li> [0m
App 31162 stderr:        </ul>
App 31162 stderr:      </div>
App 31162 stderr:    </aside>
App 31162 stderr:
App 31162 stderr:    <div class="grid__item three-fifths lap-two-thirds
palm-one-whole">
App 31162 stderr:      <p class="lead">There are several ways to create a
transcription file, but for Windows users our bespoke program WinFreeREG is
the best choice. For those on Mac or Linux, we recommend using a
spreadsheet, as found in any office software.</p>
App 31162 stderr:
App 31162 stderr:      <div class="island island--bordered">
App 31162 stderr:        <p>A FreeReg file should be no larger than about
2,000 records. We strongly recommend that those records be all for the same
Place, Church and Register type: that is, they should amount to a single
Batch.</p>
App 31162 stderr:        <p>Before you begin to work, you must ensure that
you have the necessary <a
href="/cms/information-for-transcribers/permission-to-transcribe">Permission
to transcribe?</a> for your source.</p>
App 31162 stderr:      </div>
App 31162 stderr:
App 31162 stderr:      <h2>WinFreeREG</h2>
App 31162 stderr:      <p>The new WinFreeREG is now available! You simply
use this link to <a href="
http://stash.freeukgen.org.uk/winreg/SetupWinFreeReg.1.0.1.0.msi">download
the WinFreeREG installer</a> and then run it. WinFreeREG is written and
maintained by Mike Fry, an experienced transcriber.</p>
App 31162 stderr:      <p>There are many advantages to using WinFreeREG:</p>
App 31162 stderr:      <ul>
App 31162 stderr:        <li>Your files will be <q>database ready</q>
without any of the extra work required by using other programs.</li>
App 31162 stderr:        <li>The program ensures that your file is set up
for an existing Place and Church. This is a new feature.</li>
App 31162 stderr:
App 31162 stderr:        <li>You will be asked for your details only once:
these will be put automatically into every file you create.</li>
App 31162 stderr:        <li>It is also easy to edit the other details that
vary from file to file. (These correspond to the remaining file Header
fields, if you have previously used a spreadsheet.)</li>
App 31162 stderr:        <li>The field names (column names in a
spreadsheet) will be created for you, according to whether you are
transcribing baptisms, marriages or burials.</li>
App 31162 stderr:        <li>You can rearrange the fields so that they are
in an order that follows the order of information in your source. When you
save a file, WinFreeREG will save the fields in the order that the database
expects.</li>
App 31162 stderr:        <li>You can also hide any fields that your source
does not have data for, just as you can with a spreadsheet.</li>
App 31162 stderr:      </ul>
App 31162 stderr:      <p>For details of how to use WinFreeREG, see the
program's Help menu. Context-sensitive help (using the F1 key) is available
on most input screens: some have a white on blue <q>i</q> icon that you can
click for more information.</p>
App 31162 stderr: [31m-    <div class="push-half"><a class="btn btn--small
weight--normal go" href="
http://stash.freeukgen.org.uk/winreg/SetupWinFreeReg.1.0.1.0.msi">Download
WinFreeREG</a> </div> [0m
App 31162 stderr: [32m+    <div class="push-half"> [0m
App 31162 stderr: [32m+<a class="btn btn--small weight--normal go" href="
http://stash.freeukgen.org.uk/winreg/SetupWinFreeReg.1.0.1.0.msi">Download
WinFreeREG</a> </div> [0m
App 31162 stderr:
App 31162 stderr: [31m-    <hr class="rule rule--ornament" /> [0m
App 31162 stderr: [32m+    <hr class="rule rule--ornament"> [0m
App 31162 stderr:
App 31162 stderr:      <h2 id="spreadsheet">Spreadsheet</h2>
App 31162 stderr:      <p>If you are not able to use WINFreeReg for any
reason, then we recommend using a spreadsheet, especially one from the Open
Source office suites: the first two are available for Windows, Mac OS X and
Linux; they are also free of charge.</p>
App 31162 stderr:      <ul>
App 31162 stderr: [31m-      <li><a class="go" href
App 31162 stderr: ="https://www.openoffice.org/" title="Apache OpenOffice
website">Apache OpenOffice</a> [0m
App 31162 stderr: [32m+      <li> [0m
App 31162 stderr: [32m+<a class="go" href="https://www.openoffice.org/"
title="Apache OpenOffice website">Apache OpenOffice</a> [0m
App 31162 stderr:        </li>
App 31162 stderr: [31m-      <li><a class="go" href="
https://www.libreoffice.org/" title="LibreOffice website">LibreOffice</a>
</li> [0m
App 31162 stderr: [31m-      <li><a class="go" href="
https://www.neooffice.org/neojava/en/index.php" title="NeoOffice
website">NeoOffice</a> is for Mac OS X only and there is a small charge
(starting at £8 / $10 at time of writing)</li> [0m
App 31162 stderr: [32m+      <li> [0m
App 31162 stderr: [32m+<a class="go" href="https://www.libreoffice.org/"
title="LibreOffice website">LibreOffice</a> </li> [0m
App 31162 stderr: [32m+      <li> [0m
App 31162 stderr: [32m+<a class="go" href="
https://www.neooffice.org/neojava/en/index.php" title="NeoOffice
website">NeoOffice</a> is for Mac OS X only and there is a small charge
(starting at £8 / $10 at time of writing)</li> [0m
App 31162 stderr:      </ul>
App 31162 stderr:      <p>A modern version of MS Excel can be used, but we
do not recommend older versions as they have a bug in the way they save CSV
files — the kind of file that must be uploaded to the database. There is a
way around this bug, but the fix involves extra work, and may itself
introduce a different problem.</p>
App 31162 stderr:      <p>Other proprietary spreadsheets are available, but
we have less collective experience with these.</p>
App 31162 stderr:      <p>You don't need to know how to do anything
complicated with the spreadsheet, just how to: Save, Save as, and set the
format of cells in a column (field) to Text. It would also be very useful
to know how to freeze the headings, so that they are always visible. In
addition, we ask that you <strong>turn off all the AutoCorrect
options</strong> — the details of why and how are in the box below.</p>
App 31162 stderr:      <p>We recommend that you save your working file in
the spreadsheet's own format (file ending). This will make it simpler to
re-open the file and will preserve such things as column widths and font
size for when you next work on your transcription.</p>
App 31162 stderr:
App 31162 stderr:      <div class="island island--bordered">
App 31162 stderr:        <h3>AutoCorrect</h3>
App 31162 stderr:        <p>AutoCorrect or Substitution is now a common
feature in many applications: even the so-called plain text editors like
Notepad or TextEdit are not so plain any more. The default (or out of the
box) setting is to have the feature turned <em>on</em>: this is not a
problem for many kinds of writing, but for a database project like FreeReg,
there are lurking dangers.</p>
App 31162 stderr:        <p>Many of the corrections applied by AutoCorrect
are to do with the casing of letters and the common typing errors people
make with words. AutoCorrect can ensure that all sentences start with a
capital (upper case) letter; it can prevent you beginning a word with two
capital letters; and it can change the word i to I. It can also correct
typos like abotu to about, or acn to can. This is not a problem in itself,
but for FreeReg work, you will want to be in full control of how words are
spelt and cased. The main problem is that AutoCorrect changes some plain
punctuation to their smart (fancy or typographers) versions.</p>
App 31162 stderr:        <p>A good example of AutoCorrect in action with
punctuation happens like this: you type a plain double quotation mark on
your keyboard: it starts out looking like ", but as soon as you type the
next thing, it changes magically to “ — the plain quote has been converted
to a fancy or smart quote. Again, this is not a problem for most kinds of
writing (indeed, it is useful for a party invitations or a novel), but for
FreeReg it <em>is</em> a problem.</p>
App 31162 stderr: [31m-      <p>As well as changing " to “ or ”, other
common substitutions are the single quotation mark (and apostrophe), ' to ‘
or ’; ellipses, ... (three characters) to … (one character); and
dashes,&#160;<abbr title="space hyphen space">&#160;-&#160; </abbr>&#160;or
-- to —</p> [0m
App 31162 stderr: [32m+      <p>As well as changing " to “ or ”, other
common substitutions are the single quotation mark (and apostrophe), ' to ‘
or ’; ellipses, ... (three characters) to … (one character); and dashes,
<abbr title="space hyphen space"> -  </abbr> or -- to —</p> [0m
App 31162 stderr:        <p>The exact form of the smart punctuation will
vary from font to font, but there is less difference when comparing the
plain forms. For the quotation marks, the plain forms will be vertical,
with left and right quotes looking the same. (If you are having trouble
seeing the difference, try zooming in: see your browser's View menu.)</p>
App 31162 stderr:
App 31162 stderr:        <h4>The problems</h4>
App 31162 stderr:
App 31162 stderr:        <h5>Searching</h5>
App 31162 stderr:        <p>If smart punctuation is used instead of plain,
then a search for a surname like O'Grady will not find all the possible
matches: the O'Grady entries will be found, but <em>not</em> O’Grady. The
visual difference between the two apostrophes is small, but the effect on a
search could be huge.</p>
App 31162 stderr:        <p>Similarly, most abbreviations of a forename
will be found automatically in a search, but any that include a smart
apostrophe will not be found. So Tho's will be found in a search for
Thomas, for example, but <em>not</em> Tho’s.</p>
App 31162 stderr:
App 31162 stderr:        <h5>Garbled entries</h5>
App 31162 stderr:        <p>If your file has been saved using one character
set, but the character set in the Headers is not the same, then any
<em>smart</em> punctuation will not be understood by the database and you
will see groups of characters like these in your uploaded batch: â€¦ â€œ
â€? â€˜ â€™.</p>
App 31162 stderr:        <p>This is most likely to happen when files are
passed on to someone else for editing or Uploading: the remedy is to turn
off all the AutoCorrect options, then re-type the problem punctuation.
Finally, make sure that you put the correct character set in cell 1F of the
Headers (see the <a
href="/cms/information-for-transcribers/file-headers">File Headers</a> page
for details).</p>
App 31162 stderr:
App 31162 stderr:        <h4>How to turn AutoCorrect off</h4>
App 31162 stderr: [31m-      <p>Start with a blank spreadsheet open: this
will prevent you seeing any options that do not apply to spreadsheets. From
the menu, select Tools &gt; AutoCorrect&#160;Options. You will get a dialog
box with four parts: Replace, Exceptions, Options, Custom Quotes or
Localised Options. (Note that OpenOffice, for example, has a fifth part,
Word Completion, which you will see if you do not have a blank spreadsheet
open: you can ignore it.)</p> [0m
App 31162 stderr: [32m+      <p>Start with a blank spreadsheet open: this
will prevent you seeing any options that do not apply to spreadsheets. From
the menu, select Tools &gt; AutoCorrect Options. You will get a dialog box
with four parts: Replace, Exceptions, Options, Custom Quotes or Localised
Options. (Note that OpenOffice, for example, has a fifth part, Word
Completion, which you will see if you do not have a blank spreadsheet open:
you can ignore it.)</p> [0m
App 31162 stderr:        <p>The third and fourth parts are the important
ones: the first and second contain the details of how some correction
options are applied — we are going to turn those options off.</p>
App 31162 stderr:        <p>Part three (Options) has a list of possible
corrections: simply de-select all of these.</p>
App 31162 stderr:        <figure>
App 31162 stderr: [31m-        <img style="width: 100%; height: auto;"
data-rel="225x255" alt="Part 3 of AutoCorrect in OpenOffice" title="Part 3
of AutoCorrect in OpenOffice"
src="/system/images/W1siZiIsIjIwMTYvMDcvMDQvMjJfMTVfMDBfMTcyX0F1dG9Db3JyZWN0XzMucG5nIl1d/AutoCorrect-3.png?sha=0ff1f2db53a9d372"
width="593" height="494" /> [0m
App 31162 stderr: [32m+        <img style="width: 100%; height: auto;"
data-rel="225x255" alt="Part 3 of AutoCorrect in OpenOffice" title="Part 3
of AutoCorrect in OpenOffice"
src="/system/images/W1siZiIsIjIwMTYvMDcvMDQvMjJfMTVfMDBfMTcyX0F1dG9Db3JyZWN0XzMucG5nIl1d/AutoCorrect-3.png?sha=0ff1f2db53a9d372"
width="593" height="494"> [0m
App 31162 stderr:          <figcaption>Part 3 of AutoCorrect in
OpenOffice</figcaption>
App 31162 stderr:        </figure>
App 31162 stderr:        <p>Part four (Custom Quotes or Localised Options)
deals with quotation marks: un-check both Replace boxes. Un-check anything
else that is offered here for writing in English.</p>
App 31162 stderr:        <figure>
App 31162 stderr: [31m-        <img style="width: 100%; height: auto;"
data-rel="225x255" alt="Part 4 of AutoCorrect in OpenOffice" t
App 31162 stderr: itle="Part 4 of AutoCorrect in OpenOffice"
src="/system/images/W1siZiIsIjIwMTYvMDcvMDQvMjJfMTVfMzFfNzkyX0F1dG9Db3JyZWN0XzQucG5nIl1d/AutoCorrect-4.png?sha=16b966fa51401c16"
width="595" height="493" /> [0m
App 31162 stderr: [32m+        <img style="width: 100%; height: auto;"
data-rel="225x255" alt="Part 4 of AutoCorrect in OpenOffice" title="Part 4
of AutoCorrect in OpenOffice"
src="/system/images/W1siZiIsIjIwMTYvMDcvMDQvMjJfMTVfMzFfNzkyX0F1dG9Db3JyZWN0XzQucG5nIl1d/AutoCorrect-4.png?sha=16b966fa51401c16"
width="595" height="493"> [0m
App 31162 stderr:          <figcaption>Part 4 of AutoCorrect in
OpenOffice</figcaption>
App 31162 stderr:        </figure>
App 31162 stderr:      </div>
App 31162 stderr:
App 31162 stderr:      <h3>Naming your file</h3>
App 31162 stderr:      <p class="push-half--bottom">The CSV file, described
below, that you will Upload or Replace on the server needs to be named
according to the following pattern. The first 8 letters of the name are:</p>
App 31162 stderr:      <ul>
App 31162 stderr:        <li>The 3-letter County code (as listed on the <a
href="/cms/information-for-transcribers/chapman-codes">Chapman codes</a>
page)</li>
App 31162 stderr:        <li>The 3-letter Place code (please ask your
Coordinator for this code)</li>
App 31162 stderr:        <li>A 2-letter code that describes the event type:
BA (baptisms), BU (burials) or MA (marriages)</li>
App 31162 stderr:      </ul>
App 31162 stderr:      <p>For smaller registers of around two thousand
records or less in total, these eight letters are all you need. However, if
you are going to be transcribing a larger register of more than two
thousand records, then you will need to append a number to each filename to
distinguish each part of the transcription.</p>
App 31162 stderr:      <p>For example, a baptism register of about six
thousand records for a (fictitious) Place with code ABC in the West Riding
of Yorkshire will need three files: WRYABCBA1, WRYABCBA2, and WRYABCBA3.</p>
App 31162 stderr:      <p>If the register is large enough to need more than
nine files to complete, then consider using two digits to number your
files: this will result in a set of batches that sorts in a more natural
order in batch listings. This is because batches are sorted in so-called
dictionary order, so that WRYABCBA2, for instance, would be listed after
WRYABCBA12, whereas WRYABCBA<abbr title="This is a zero">0</abbr>2 would be
listed before it.</p>
App 31162 stderr:
App 31162 stderr:      <h3>The final step</h3>
App 31162 stderr:      <p>There is a final step which should be done in
your spreadsheet application: adding the information that lets the database
understand your transcription and know who created it. The instructions are
on the <a href="/cms/information-for-transcribers/file-headers">File
headers</a> page. If you have any problems with this step, your Syndicate
Coordinator will be pleased to help you.</p>
App 31162 stderr:
App 31162 stderr:    </div>
App 31162 stderr:  </div>
App 31162 stderr:  <div class="grid">
App 31162 stderr:    <div class="grid__item three-fifths lap-two-thirds
palm-one-whole">
App 31162 stderr:      <h3>Save as … CSV</h3>
App 31162 stderr:      <p>When you have finished entering data, the next
step is to save your file in <abbr title="Comma-separated
values">CSV</abbr> format. Just before doing so, delete the row of column
labels, if you haven't already deleted them as part of adding the file
Headers.</p>
App 31162 stderr:      <p>With your spreadsheet open, use File, Save As to
bring up a dialog box. Choose Comma-separated values as the format,
probably from a drop-down list. You may be asked to confirm (or choose)
commas as the <q>field separators</q> (or <q>field delimiters</q>), and
double-quotes as the <q>text delimiters</q>: agree to both. It is not
necessary to agree to <em>all</em> fields being quoted, but it will do no
harm.</p>
App 31162 stderr:      <p>A CSV file just a simple text file, with a .csv
or .CSV extension (ending). Each data field (cell) is separated from the
next by a comma, and each record is on its own line. For a baptism
register, opened in a text editor such as Notepad or TextEdit (not a word
processor), the beginning of the first two lines after the Headers might
look something like this:</p>
App 31162 stderr:    </div>
App 31162 stderr:
App 31162 stderr:    <div class="grid__item one-whole">
App 31162 stderr: [31m-    <code>GLS,Ruardean,All Saints PR,,,06 Jun
1786,Elizabeth,F, … <br /> [0m
App 31162 stderr: [32m+    <code>GLS,Ruardean,All Saints PR,,,06 Jun
1786,Elizabeth,F, … <br> [0m
App 31162 stderr:            GLS,Ruardean,All Saints PR,,,20 Jun
1786,James,M, … </code>
App 31162 stderr:    </div>
App 31162 stderr:
App 31162 stderr:    <div class="grid__item three-fifths lap-two-thirds
palm-one-whole">
App 31162 stderr:      <p class="push--top">Notice that some of the fields
are empty in this example, with literally nothing between the commas: there
are no unnecessary spaces. </p>
App 31162 stderr:
App 31162 stderr:      <h3 id="first_time_check">First time check</h3>
App 31162 stderr:      <p>This test is not necessary for users of Apache
OpenOffice, LibreOffice or NeoOffice — all three pass.</p>
App 31162 stderr:      <p>If this is your first transcription file with MS
Excel (pre-2011) or another program not listed above, then this simple
check will tell you whether or not it is suitable for FreeReg: in
particular, some of the older versions of Excel have a bug in the way they
save CSV files that makes them unsuitable. </p>
App 31162 stderr:
App 31162 stderr:      <p>The check is simply this:</p>
App 31162 stderr:
App 31162 stderr:      <ol>
App 31162 stderr:        <li>Open the CSV file in a text editor (Notepad or
TextEdit)</li>
App 31162 stderr:        <li>Find the end of a line that you know does
<em>not</em> have data in the final field (column):
App 31162 stderr:          <ul>
App 31162 stderr:            <li>If it ends with one or more commas, all is
well!</li>
App 31162 stderr:            <li>If there are no commas at the end, then
you will need to use one of the free Office programs</li>
App 31162 stderr:          </ul>
App 31162 stderr:        </li>
App 31162 stderr:      </ol>
App 31162 stderr:
App 31162 stderr:      <p>Not sure? Some examples may help. A baptism entry
<strong>with</strong> data in the final field — in this case, the Notes
field — will end something like this, with no final comma:</p>
App 31162 stderr:
App 31162 stderr:      <p><code>… ,WARD,FORBES,"23, High St",,Mr Saml
Ward</code>
App 31162 stderr:      </p>
App 31162 stderr:
App 31162 stderr:      <p>(Note also that one of the fields has double
quotes around the contents: this is because the contents — 23, High St —
has a comma in it. Fields without a comma are not quoted.)</p>
App 31162 stderr:
App 31162 stderr:      <p>Lines <strong>without</strong> data in the final
field, should end something like these:</p>
App 31162 stderr:
App 31162 stderr: [31m-    <p><code>… ,BAIRD,,,, <br /> [0m
App 31162 stderr: [32m+    <p><code>… ,BAIRD,,,, <br> [0m
App 31162 stderr:      … ,REYNOLDES,JONES,,Blacksmith,</code> </p>
App 31162 stderr:
App 31162 stderr:      <p>A line <strong>without</strong> content in the
final field should end with <em>at least one</em> comma. If you see one or
more commas at the end of such lines, then your spreadsheet has been saved
correctly. <strong>You will not need to do this check again.</strong>
App 31162 stderr:      </p>
App 31162 stderr:
App 31162 stderr:      <p>However, if you’re seeing entries like these,
with <em>empty</em> final fields but <em>no</em> final comma(s),  </p>
App 31162 stderr:
App 31162 stderr: [31m-    <p><code>… ,BAIRD <br /> [0m
App 31162 stderr: [32m+    <p><code>… ,BAIRD <br> [0m
App 31162 stderr:      … ,REYNOLDES,JONES,,Blacksmith</code> </p>
App 31162 stderr:
App 31162 stderr:      <p>then you will need to download one of the free
Office suites for your FreeReg transcriptions. (Experienced volunteers who
use old versions of MS Excel and the "dummy column" method without any
problems may continue to do so, but we urge you to consider using a
different spreadsheet, if possible.)</p>
App 31162 stderr:
App 31162 stderr: [31m-    <hr class="rule rule--ornament" /> [0m
App 31162 stderr: [32m+    <hr class="rule rule--ornament"> [0m
App 31162 stderr:
App 31162 stderr:      <h2 id="database">Database</h2>
App 31162 stderr:      <p>If you are more comfortable with database
software, then any that can save files in the required "Comma-separated
values" (CSV) format will be suitable. You will need to add the headers to
your completed CSV file, as described for spreadsheets. If your software
has AutoCorrect or Substitution options, you need to turn off all that
apply to punctuation.</p>
App 31162 stderr:
App 31162 stderr: [31m-    <hr class="rule rule--ornament" /> [0m
App 31162 stderr: [32m+    <hr class="rule rule--ornament"> [0m
App 31162 stderr:
App 31162 stderr:      <h2 id="textEditor">Text editor</h2>
App 31162 stderr:      <p>This method for FreeReg transcription is really
only for people who use a text editor for everything and wouldn't dream of
using anything else. You would have to set everything up yourself, be good
at counting commas and remember to quote fields that themselves contain
commas. (Full details can be found on <a
href="/cms/information-for-transcribers/entering-data-from-registers">Entering
data from registers</a>.)</p>
App 31162 stderr:      <p>You would also have to enter the file Headers
yourself, as described on <a
href="/cms/information-for-transcribers/file-headers">File headers</a>.</p>
App 31162 stderr:      <p>Plain text editors are no longer quite so plain.
In par
App 31162 stderr: ticular, they may have an AutoCorrect or Substitution
feature: you need to <em>turn off</em> all substitutions that apply to
punctuation, especially for quotation marks, ellipses and hyphens/dashes.
This applies even if you are just using a text editor to make a small edit
of your file or someone else's file.</p>
App 31162 stderr:      <p>Please check that you know which character set
your editor uses on saving: if offered a choice, then select utf-8 and put
UTF-8 in line 1, field 6 (character set) of the file Headers.</p>
App 31162 stderr:
App 31162 stderr:    </div>
App 31162 stderr:  </div>
App 31162 stderr:
App 714 stdout:   Rendered
vendor/bundle/ruby/2.2/gems/refinerycms-core-3.0.4/app/views/refinery/_content_page.html.erb
(134.1ms)
App 714 stdout:   Rendered
vendor/bundle/ruby/2.2/gems/refinerycms-pages-3.0.4/app/views/refinery/pages/show.html.erb
within layouts/application (134.4ms)
App 714 stdout:   Rendered layouts/_main_nav.html.erb (0.5ms)
App 714 stdout:   Rendered layouts/_header.html.erb (0.2ms)
App 714 stdout:   Rendered layouts/_footer.html.erb (0.6ms)
App 714 stdout: Completed 200 OK in 150ms (Views: 137.8ms | ActiveRecord:
2.2ms)
App 31799 stdout: Redirected to
http://www.freereg.org.uk/search_queries/586abd48f493fd7c37968d47
App 31799 stdout: Completed 302 Found in 746ms (ActiveRecord: 0.0ms)
App 31799 stdout: Started GET
"/assets/styles/css/palm.min-3f1d89d525404456b26aeb03da3eadc6304db55af5b99885a041ea14d603229c.css"
for 5.9.125.163 at 2017-01-02 20:51:21 +0000
App 31799 stdout:
App 31799 stdout: ActionController::RoutingError (No route matches [GET]
"/assets/styles/css/palm.min-3f1d89d525404456b26aeb03da3eadc6304db55af5b99885a041ea14d603229c.css"):

Kirk Dawson
5220 Riverside Drive
Fairmont Hot Springs, B.C.
V0B 1L1

**Comment by Captainkirkdawson** [comment]
2 remaining issues
Refinery assets still missing
Some css still missing

**Comment by richpomfret** [comment]
To attempt to replicate.

**Comment by Captainkirkdawson** [comment]
I have found the cause. The public folder and all its contents had been moved into the lib folder,  They have been moved back to where they should be and the system on colobus is now running as it was previously. Nothing more needs to be done.
What was the cause. Clearly nothing to do with Refinery. Given the time it was most likely me. I use a gui to view the contents of the server. This enables me to download and upload files. Mostly of the logs and reports. I does have the ability for me to rename and drag and drop files or folders. Most of these such actions the server will not respond to because I do not have the correct permission and I cannot bypass the permission with a sudo. However I found in "retrieving the public folder" that the gui would move under different permission settings.

Bottom line it was my error. I need to be more careful with the gui
